cylc/flow/task_proxy.py

```diff
@@ -236,7 +236,10 @@ def __init__(
 
         self.local_job_file_path: Optional[str] = None
 
-        self.platform = None
+        if data_mode:
+            self.platform = {}
+        else:
+            self.platform = None
```

Comment:
This looks redundant to me. I think it should be either `None` or `{}` in any case

Reply:
I agree.